packages/lib/server/repository/schedule.ts

```diff
@@ -85,4 +85,86 @@ export class ScheduleRepository {
       readOnly: schedule.userId !== userId && !isManagedEventType,
     };
   }
+
+  static async getAllDetailedScheduleByUserId({
+    isManagedEventType,
+    timeZone,
+    currentUserId,
+    userIdOfSchedulesToGet,
+  }: {
+    timeZone: string;
+    currentUserId: number;
+    isManagedEventType?: boolean;
+    userIdOfSchedulesToGet: number;
+  }) {
+    const isCurrentUserPartOfTeam = hasReadPermissionsForUserId({
+      memberId: userIdOfSchedulesToGet,
+      userId: currentUserId,
+    });
+
+    const isCurrentUserOwner = userIdOfSchedulesToGet === currentUserId;
+
+    if (!isCurrentUserPartOfTeam && !isCurrentUserOwner) {
+      throw new Error("UNAUTHORIZED");
+    }
+
+    const foundUserDefaultId = await prisma.user.findUnique({
+      where: {
+        id: userIdOfSchedulesToGet,
+      },
+      select: {
+        defaultScheduleId: true,
+      },
+    });
+
+    if (!foundUserDefaultId?.defaultScheduleId) {
+      const EMPTY_SCHEDULE = [[], [], [], [], [], [], []];
+      return [
+        {
+          id: -1,
+          name: "Working Hours",
+          availability: EMPTY_SCHEDULE,
+          dateOverrides: [],
+          timeZone: timeZone || "Europe/London",
+          workingHours: [],
+          isDefault: true,
+          hasDefaultSchedule: false, // This is the path that we take if the user has not completed onboarding
+        },
+      ];
+    }
```

Comment:
```ts
const defaultScheduleId = await getDefaultScheduleId(input.userId, prisma);
```

Earlier this would return the first schedule it found in case user did not have any default schedule. Replaced it with the logic to actually check if user has completed onboarding. 

          hasDefaultSchedule: false, // This is the path that we take if the user has not completed onboarding
          
          This flag is used to determine state for frontend